src/core/meshcop/border_agent.cpp

```diff
@@ -308,21 +309,38 @@ Error BorderAgent::SetId(const Id &aId)
 
 void BorderAgent::HandleNotifierEvents(Events aEvents)
 {
-    VerifyOrExit(aEvents.ContainsAny(kEventThreadRoleChanged | kEventCommissionerStateChanged));
-
+    if (aEvents.ContainsAny(kEventThreadRoleChanged | kEventCommissionerStateChanged))
+    {
 #if OPENTHREAD_CONFIG_COMMISSIONER_ENABLE && OPENTHREAD_FTD
-    VerifyOrExit(Get<Commissioner>().IsDisabled());
+        VerifyOrExit(Get<Commissioner>().IsDisabled());
 #endif
 
-    if (Get<Mle::MleRouter>().IsAttached())
-    {
-        Start();
+        if (Get<Mle::MleRouter>().IsAttached())
+        {
+            Start();
+        }
+        else
+        {
+            Stop();
+        }
     }
-    else
+
+    if (aEvents.ContainsAny(kEventPskcChanged))
     {
-        Stop();
+        if (mState == kStateStarted)
+        {
+            Restart();
+        }
+
+        mPskcChanged = true;
+
+        // If there's an active commissioner or candidate in `kStateConnected`
+        // or `kStateAccepted`, maintain the connection. The BA will restart and
+        // use the new PSKc once the connection is closed.
```

Comment:
Is it safe to change the `psk` without restarting? From `BorderAgent::Start`, the psk is set after SecureAgent is started.

Reply:
`Restart()` will be executed if there is pskc change, immediately or later when connection is closed.

It should be fine as long as pskc is set into secure transport before connection (mbedtls setup where applying pskc)

https://github.com/openthread/openthread/blob/45c5fe475b281b80c426f2e9e80433dc5992a403/src/core/meshcop/secure_transport.cpp#L513


https://github.com/openthread/openthread/blob/45c5fe475b281b80c426f2e9e80433dc5992a403/src/core/meshcop/secure_transport.cpp#L378

Reply:
I meant is it necessary to wait for the connection to close? Is it safe to just set the new psk? The secure transport should take the new value when it's preparing for a new connection.